### PR TITLE
Improve integration test coverage for role package

### PR DESCRIPTION
**Pull request by sajitha-tj**
### Purpose
Improve integration test coverage for the `backend/internal/role` package.

### Approach
Added `tests/integration/role/roleapi_test.go` extensions and a new `tests/integration/role/role_authz_test.go` with `TestRoleAuthzTestSuite`, plus a supporting `model.go` for role test fixtures. `TestRoleAPITestSuite` now covers 67 top-level cases (previously a smaller set), adding error paths, filter/pagination edge cases, dependency-delete guards, and cross-store declarative role behavior for `/roles`.

Measured integration coverage after this change:

| Package / file | Coverage | Statements |
|---|---|---|
| `internal/role` | 79.1% | 1552/1962 |

Remaining uncovered paths are mostly error/unreachable branches: the file-based store's stub cascade-delete paths, the composite store's `IsRoleDeclarative` variant in `store.go`, and `ResolveForRole` in `permission_resolver.go`.

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [x] Followed the contribution guidelines.
- [x] Manual test round performed and verified.
- [ ] Documentation provided. (Add links if there are any)
    - [ ] Ran Vale and fixed all errors and warnings
- [x] Tests provided. (Add links if there are any)
    - [ ] Unit Tests
    - [x] Integration Tests
- [ ] Breaking changes. (Fill if applicable)
    - [ ] Breaking changes section filled.
    - [ ] `breaking change` label added.

### Security checks
- [x] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

## Summary by CodeRabbit

- **New Features**
  - Added support for assigning roles to agents, users, and groups.
  - Added organization-unit role listings with pagination.
  - Added role export and access-evaluation capabilities.
  - Added declarative role permissions, assignments, and organization-unit references.

- **Bug Fixes**
  - Improved validation for malformed requests, invalid assignments, name conflicts, missing organization units, and routing errors.
  - Enforced authorization requirements for role management and privileged group membership changes.

- **Tests**
  - Expanded integration coverage for role assignments, declarative roles, exports, access evaluation, pagination, CORS, and authorization behavior.

### CODE REVIEW

File: tests/integration/role/roleapi_test.go [hunk 1]
```diff
@@ -1577,6 +1594,690 @@ func (suite *RoleAPITestSuite) TestDeleteResourceServer_CascadesToRolePermission
...
+	suite.Require().NoError(err)
+	suite.Equal(2, page1.TotalResults, "TotalResults must reflect the group-filtered count")
+	suite.Require().Equal(1, page1.Count)
+
+	page2, err := suite.getRoleAssignmentsByType(role.ID, 1, 1, "group")
+	suite.Require().NoError(err)
+	suite.Equal(2, page2.TotalResults)
+	suite.Require().Equal(1, page2.Count)
+	suite.NotEqual(page1.Assignments[0].ID, page2.Assignments[0].ID,
+		"the two pages must return different groups")
```
[the remark this answers is not in the document]
Comment (sajitha-tj) [reply tests/integration/role/roleapi_test.go:2072]: I am working only on improving integration tests for this package and the existing method validates pagination in a fare way. Backend orders assignments only by `CREATED_AT`. This fix should come from backend if needed.

**Review by senthalan — APPROVED** [review]
